src/main/java/org/openstreetmap/atlas/geography/sharding/DynamicTileSharding.java

```diff
@@ -114,6 +118,50 @@ public boolean equals(final Object other)
             return false;
         }
 
+        /**
+         * Does a deep equals with the other node
+         * 
+         * @param other
+         *            other Node
+         * @return true if entire structure is equal, false if not
+         */
+        private boolean deepEquals(final Node other)
+        {
+            final Comparator<Node> nodeCompare = Comparator.comparing(Node::getTile);
+            // BFS through both trees to get equality
+            final Queue<Node> queue = new LinkedList<>();
+            queue.offer(this);
+            queue.offer(other);
+            while (!queue.isEmpty())
+            {
+                // We always offer two at a time, so we can poll two at a time.
+                final Node node1 = queue.poll();
+                final Node node2 = queue.poll();
+                if (node1.equals(node2) && node1.getChildren().size() == node2.getChildren().size())
+                {
+                    final List<Node> children1 = node1.getChildren();
+                    final List<Node> children2 = node2.getChildren();
+                    children1.sort(nodeCompare);
+                    children2.sort(nodeCompare);
+                    for (int itr = 0; itr < children1.size(); itr++)
```

Comment:
nit: either make this `i` or spell out the abbreviation

Reply:
can't make it i for check style I believe

Reply:
Checkstyle should allow one letter variable names in loops, per the config we currently have:
```
<module name="LocalVariableName" >
                        <property name="format" value="^([a-z][a-zA-Z0-9]{2,}$|[e])$"/>
                        <property name="allowOneCharVarInForLoop" value="true"/>
</module>
```
If this is not the case (and you are seeing Checkstyle failures) please let us know because that is a potential config issue on our part we will need to address.

Reply:
I am going to merge this for now since it's not a huge deal - but again I would like to know if using `i` is causing your `checkstyleMain` to fail because that would be a config issue on our part.